### ISSUE TITLE
[BUG]:  "zenml example pull" still refers to an older release

### ISSUE THREAD

**Issue by tunayokumus** [zenml-io/zenml#772]
### System Information

ZenML version: 0.10.0
Install path: /home/t00238/miniconda3/envs/zenml_v0_10/lib/python3.8/site-packages/zenml
Python version: 3.8.13
Platform information: {'os': 'linux', 'linux_distro': 'ubuntu', 'linux_distro_like': 'debian', 'linux_distro_version': '20.04'}
Environment: native
Integrations: ['great_expectations', 'scipy', 'sklearn']

### What happened?

`zenml example pull` is not able to check out to the latest release branch. 
I also have other instances of zenml installed in different virtual environment. Not sure if there is data leakage from existing zenml installations to this one. Do they share any user level files? Or each zenml installation is isolated in their virtual environment completely?

### Reproduction steps

1. Have multiple zenml installations in various venvs.
2. Follow the instructions on https://github.com/zenml-io/zenml/tree/main/examples/great_expectations_data_validation

### Relevant log output

```shell
> zenml example pull great_expectations_data_validation 

Checking out branch: release/0.10.0
The specified branch release/0.10.0 not found in repo, falling back to the latest release.
Checking out branch: release/0.6.0
Error: "Example great_expectations_data_validation does not exist! Available examples: ['airflow_local', 'caching', 'class_based_api', 'custom_materializer', 'dag_visualizer', 'drift_detection', 'fetch_historical_runs', 'functional_api', 'kubeflow', 'lineage', 'mlflow_tracking', 'not_so_quickstart', 'quickstart', 'standard_interfaces', 'statistics', 'whylogs']"
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by stefannica** [comment]
@tunayokumus thank you for filing this !

I think yours might be a case of cache deprecation. ZenML does keep a cache of all examples it downloads locally in [the global configuration directory](https://docs.zenml.io/developer-guide/repo-and-config#the-global-config) under a `zenml_examples` subfolder.

You can correct the problem by simply passing the `-y` option when you run `zenml example pull`:

```shell
zenml example pull -y great_expectations_data_validation
```
```

**Comment by fa9r** [comment]
Fixed the underlying issue in #776
